iroh/src/node/rpc.rs

```diff
@@ -842,6 +937,120 @@ impl<D: BaoStore> Handler<D> {
         })
     }
 
+    fn batch_create(
+        self,
+        _: BatchCreateRequest,
+        mut updates: impl Stream<Item = BatchUpdate> + Send + Unpin + 'static,
+    ) -> impl Stream<Item = BatchCreateResponse> {
+        let batch = self.inner.blob_batches.lock().unwrap().create();
+        tokio::spawn(async move {
+            while let Some(item) = updates.next().await {
+                match item {
+                    BatchUpdate::Drop(content) => {
+                        // this can not fail, since we keep the batch alive.
+                        // therefore it is safe to ignore the result.
+                        let _ = self.inner.blob_batches.lock().unwrap().remove_one(
+                            batch,
+                            &content,
+                            self.inner.db.tag_drop(),
+                        );
+                    }
+                    BatchUpdate::Ping => {}
+                }
+            }
+            self.inner
+                .blob_batches
+                .lock()
```

Comment:
if this is mostly used in an async context, should be an async lockd

Reply:
Using a sync lock in an async context is fine provided you don't lock over an await point and the lock is held for a very short time. Both is the case here.

Reply:
I know, but in the context of the RPC, this is still risky, because instead of yielding, you block the thread, and there is really no benefit of using a standard mutex in an async context unless you are in a super hot path

Reply:
I don't see the issue.

Most operations (create and store) are extremely cheap and constant time. remove_one is also cheap, it just decrements a counter in a map. The only op that is even somewhat expensive is remove, and that just decrements a bunch of counters.

So if for some reason you have an absolutely giant number of temp tags in a batch, and then you call remove for that batch, the lock will be taken during that time. And if you then do any op that also relates to batches, you will block an executor for a millisecond.

But ok, in this case it is not super painful to use the async lock, so I will give it a try.